amlip_cpp/src/cpp/types/GenericType.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file GenericType.cpp
+ * This source file contains the definition of a generic type that contains void* data.
+ */
+
+#ifdef _WINID_SIZE
+// Remove linker warning LNK4221 on Visual Studio
+namespace {
+char dummy;
+}  // namespace
+#endif  // _WINID_SIZE
+
+#include <fastcdr/Cdr.h>
+
+#include <fastcdr/exceptions/BadParamException.h>
+using namespace eprosima::fastcdr::exception;
+
+#include <array>
+#include <stdlib.h>
+
+#include <types/GenericType.hpp>
+
+namespace eprosima {
+namespace amlip {
+namespace types {
+
+GenericType::GenericType(
+        void* data,
+        const uint32_t size)
+    : data_(data)
+    , data_size_(size)
+    , has_been_allocated_(false)
+{
+}
+
+GenericType::GenericType()
+    : GenericType(nullptr, 0)
+{
+}
+
+GenericType::~GenericType()
+{
+    // In case the data has been allocated from this class, we free it.
+    if (has_been_allocated_)
+    {
+        free(data_);
+    }
+}
+
+GenericType::GenericType(
+        const GenericType& x)
+{
+    data_ = x.data_;
```

Comment:
Again the problem about if this should "copy" or "deepcopy" the data.
Don't know the best way to do it, I just want to open the discussion.

Reply:
It's been decided to copy the pointer and not the content.

Reply:
Finally it has been decided to remove the copy constructor and copy assignment operator, as it is not totally clear which would be the best option.

Reply:
Not completely sure about it, but it may be necessary to explicitly specify that these methods are deleted: `GenericDataType(const GenericDataType& x) = delete;`